src/utils/filter-store.ts

```diff
@@ -0,0 +1,32 @@
+import { FilterType } from '../models/filter';
+
+type Listener = (_filter: FilterType) => void;
+
+const getInitialFilter = (): FilterType => {
+  const urlFilter = new URLSearchParams(location.search).get('filter');
+  return ['all', 'active', 'completed'].indexOf(urlFilter as string) > -1 ? (urlFilter as FilterType) : 'all';
+};
```

Comment:
함수의 목적이
1. 현재 url의 filterType이 유효한지 확인
2. 유효하지 않은 경우 내부에서는 default인 'all'로 사용될 수 있도록 보정
하는 것으로 보입니다.

1의 목적으로 생각했을 때,
todo-list-app.ts의 'popstate' addEventListener 내부에서도 값을 조회하고 디폴트로 'all'을 넣어주려는 시도가 있어 아래와 같은 유틸 함수를 추가해서 공동 사용하는건 어떠실까요?
키워드를 상수로 사용하시는건 어떨지 제안드렸던 코멘트가 있어서 상수도 추가해보았습니다.
상수 위치는 따로 디렉토리를 만들어야 하겠지만, 편의상 아래에 같이 첨부했습니다!

```suggestion
export const FILTER_TYPE = {
  ALL: 'all',
  ACTIVE: 'active',
  COMPLETED: 'completed'
} as const;

const getSearchFilter = (): FilterType => {
  const urlFilter = new URLSearchParams(location.search).get('filter') ?? FILTER_TYPE.ALL;

  return (Object.values(FILTER_TYPE) as string[]).includes(urlFilter) ? (urlFilter as FilterType) : FILTER_TYPE.ALL;
};
```

2의 목적도 같이 생각해보면,
위와 같은 유틸함수 추가한 뒤 유효하지 않으면 search쪽 filterType을 강제로 'all'로 바꾸는 로직도 추가되면 더 좋을 것 같은데 어떠실까요? ㅎㅎ